src/lib.rs

```diff
@@ -10,12 +10,16 @@ use std::io::BufReader;
 use std::io::Read;
 use std::path::Path;
 
+/// This function takes a reference to a file to be parsed. If the file can not be parsed a [ParsingError] is returned.
+/// # Errors
+/// ParsingError::ParseDirectory will occur is a directory is passed. Use [parse_dir()]
 pub fn parse<T: std::convert::AsRef<std::ffi::OsStr>>(
     path: T,
 ) -> Result<aggregate_report::feedback, ParsingError> {
     let path = std::path::Path::new(&path);
     if path.is_dir() {
-        panic!("path is a directory")
+        let path_str = path.to_string_lossy().to_string();
```

Comment:
Is there a particular reason we can't just use the string with the unicode in the Err below?